core/unpack/unpacker.go

```diff
@@ -190,7 +218,7 @@ func (u *Unpacker) Unpack(h images.Handler) images.Handler {
 				span.SetAttributes(
 					tracing.Attribute("descriptor.child."+strconv.Itoa(i), []string{child.MediaType, child.Digest.String()}),
 				)
-				if images.IsLayerType(child.MediaType) {
+				if images.IsLayerType(child.MediaType) || (layerTypes != nil && layerTypes[child.MediaType]) {
```

Comment:
```suggestion
				if images.IsLayerType(child.MediaType) || layerTypes[child.MediaType] {
```

You can read from a nil map, but cannot write to it.

> A nil map is equivalent to an empty map except that no elements may be added.

https://go.dev/play/p/QB1pa8tgte-?v=goprev

Reply:
Good call, updated